### PR TITLE
Support nested `TypeVar`s

### PR DESCRIPTION
**Pull request by KotlinIsland**
eg:
```py
T = TypeVar(bound=int)
L = TypeVar(bound=list[T])
```
This PR is just for `bound`s, constraints can come later.

### CODE REVIEW

File: test-data/unit/check-based-nested-typevar.test [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+from typing import TypeVar, List
+
+T = TypeVar("T", bound=int)
+L = TypeVar("L", bound=List[T])
+
+def foo(l: L) -> T:
+    return l[0]
+
+reveal_type(foo([True]))  # N: Revealed type is "bool"
+foo([""])  # E: Value of type variable "T" of "foo" cannot be "str"  [type-var]
```
Comment (DetachHead) [inline test-data/unit/check-based-nested-typevar.test:11]: does it work on constraints and paramspecs

**Comment by DetachHead** [comment]
> TODO: constraints

in this PR or a seperate one? is there an issue for ir

File: test-data/unit/check-based-nested-typevar.test (hunk 1 above)
Comment (KotlinIsland) [reply test-data/unit/check-based-nested-typevar.test:11]: Haven't tried

**Comment by KotlinIsland** [comment]
> > TODO: constraints
> 
> in this PR or a seperate one? is there an issue for ir

sorry about that, I've got some eggs of constraints and I'll working though them now.

**Comment by KotlinIsland** [comment]
Actually, it's MVP now, constraints might be much harder.

**Review by DetachHead — APPROVED** [review]

File: test-data/unit/check-based-nested-typevar.test (hunk 1 above)
Comment (KotlinIsland) [reply test-data/unit/check-based-nested-typevar.test:11]: no

**Comment by KotlinIsland** [comment]
The primer diff is due to some new calls that are simplifying the union in the upper bound, i.e. `object | int` becomes `object`.

**Pull request merged**